### PR TITLE
Zabbix: zabbix_host: fix #65304

### LINKED ISSUE #65304: zabbix_host fails: zabbix_host.py .. KeyError: 'inventory_mode'
##### SUMMARY
The module zabbix_host [fails](https://travis-ci.org/robertdebock/ansible-role-zabbix_web/jobs/617545189#L653) using:

```
ansible_loop_var": "item", "changed": false, "item": {"description": "Example server 1 description", "groups": ["Linux servers"], "interface_dns": "server1.example.com", "interface_ip": "192.168.127.127", "link_templates": ["Template OS Linux by Zabbix agent"], "name": "Example server 1", "visible_name": "Example server 1 name"
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
zabbix_host

##### ANSIBLE VERSION
```
$ ansible --version
ansible 2.9.1
  config file = None
  configured module search path = ['/home/robertdb/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.5 (default, Oct 17 2019, 12:16:48) [GCC 9.2.1 20190827 (Red Hat 9.2.1-1)]
```

##### CONFIGURATION
```
$ ansible-config dump --only-changed
# NO OUTPUT
```

##### OS / ENVIRONMENT
Debian
Centos-7
Centos-8
Ubuntu

##### STEPS TO REPRODUCE
Git clone my [zabbix_web Ansible role](https://github.com/robertdebock/ansible-role-zabbix_web) and run `molecule test`

```
git clone https://github.com/robertdebock/ansible-role-zabbix_web.git
cd ansible-role-zabbix_web
molecule test
```

##### EXPECTED RESULTS
I was not expecting an error. The host is actually added to Zabbix by the way.

##### ACTUAL RESULTS

```
    TASK [ansible-role-zabbix_web : add zabbix hosts] ******************************
    An exception occurred during task execution. To see the full traceback, use -vvv. The error was: KeyError: 'inventory_mode'
    failed: [zabbix_web-centos-latest] (item=Example server 1) => {"ansible_loop_var": "item", "changed": false, "item": {"description": "Example server 1 description", "groups": ["Linux servers"], "interface_dns": "server1.example.com", "interface_ip": "192.168.127.127", "link_templates": ["Template OS Linux by Zabbix agent"], "name": "Example server 1", "visible_name": "Example server 1 name"}, "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1574832740.9065554-150815383322675/AnsiballZ_zabbix_host.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1574832740.9065554-150815383322675/AnsiballZ_zabbix_host.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1574832740.9065554-150815383322675/AnsiballZ_zabbix_host.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible.modules.monitoring.zabbix.zabbix_host', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib64/python3.6/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.6/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/usr/lib64/python3.6/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_zabbix_host_payload_xl5geyrr/ansible_zabbix_host_payload.zip/ansible/modules/monitoring/zabbix/zabbix_host.py\", line 902, in <module>\n  File \"/tmp/ansible_zabbix_host_payload_xl5geyrr/ansible_zabbix_host_payload.zip/ansible/modules/monitoring/zabbix/zabbix_host.py\", line 862, in main\n  File \"/tmp/ansible_zabbix_host_payload_xl5geyrr/ansible_zabbix_host_payload.zip/ansible/modules/monitoring/zabbix/zabbix_host.py\", line 546, in check_all_properties\nKeyError: 'inventory_mode'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

### PR DESCRIPTION
**Pull request by sky-joker**
##### SUMMARY
Since the position of the `inventory_mode` key has changed from zabbix 4.4.1, use `LooseVersion` to separate the processing.
Fixes https://github.com/ansible/ansible/issues/65304

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
zabbix_host

##### ADDITIONAL INFORMATION
tested on zabbix 3.0,3.2,3.4,4.0,4.2,4.4

### CODE REVIEW

**Review by D3DeFi — APPROVED** [review]
Looks good to me!

Once it is no longer WIP we can shipit

**Comment by D3DeFi** [comment]
Thanks a lot for being very helpful and active contributor @sky-joker ! Are you, by any chance, interested in maintaining other zabbix modules alongside me and @rubentsirunyan (not seen him for a while now) ?

**Comment by sky-joker** [comment]
Thank you @D3DeFi for the review!

> Thanks a lot for being very helpful and active contributor @sky-joker ! Are you, by any chance, interested in maintaining other zabbix modules alongside me and @rubentsirunyan (not seen him for a while now) ?

yes :)
I am interested in zabbix module maintaining.

**Comment by D3DeFi** [comment]
Looking forward to it! :) #65446 

I don't have always time to implement feature requests or bugfixes (and some of them are stale for some time now :( ), but I am usually responding to comments asap so don't hesitate to link me to a PR or issue anytime :)

I will try to iterate through the open PRs and issues, hopefully tomorrow, and do some housekeeping (tag ones waiting for contributor/review/etc..)

**Review by robertdebock — APPROVED** [review]
I've tested the code on my laptop and **it works**, thanks!

**Comment by sky-joker** [comment]
@D3DeFi 

> Looking forward to it! :) #65446

Thank you :)

**Comment by sky-joker** [comment]
Removed `WIP` from title!

**Review by robertdebock — APPROVED** [review]
Cool, did not know about the changelog. The entry looks complete to me.

**Comment by D3DeFi** [comment]
shipit

File: changelogs/fragments/65304-fix_zabbix_host_inventory_mode_key_error.yml
Comment (D3DeFi) [inline changelogs/fragments/65304-fix_zabbix_host_inventory_mode_key_error.yml]: On a second thought, this is at least missing module name

```suggestion
  - zabbix_host - fixed inventory_mode key error, which occurs with Zabbix 4.4.1 or more (https://github.com/ansible/ansible/issues/65304).
```

**Review by D3DeFi — CHANGES_REQUESTED** [review]

File: changelogs/fragments/65304-fix_zabbix_host_inventory_mode_key_error.yml
Comment (sky-joker) [reply changelogs/fragments/65304-fix_zabbix_host_inventory_mode_key_error.yml]: Thank you for the pointed out.
Fixed this.

**Comment by D3DeFi** [comment]
Thanks @sky-joker ! 
shipit

**Review by D3DeFi — APPROVED** [review]

**Comment by D3DeFi** [comment]
bot_status

**Comment by D3DeFi** [comment]
hmm, not sure why it is still tagged as failing. Can you try [rebasing?](https://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html)

**Comment by D3DeFi** [comment]
shipit

**Comment by D3DeFi** [comment]
bot_status

**Comment by D3DeFi** [comment]
reopening and rebasing PR didn't help obviously :( Not sure if I am right, but I would say that `bot_broken` maybe? Latest commit passed all the tests so ansibot should register this PR as NOT failed.

I think that maybe `rebuild_merge` will help, but that ansibot command can be only issued by core devs. 

@Akasurde, is there a way for me or @sky-joker to get this moving without your help other than closing this PR completely and opening a new one?

**Comment by Akasurde** [comment]
@mkrizek Do you know why bot didn't merge this PR? 
It says
 
```
automerge: automerge !module file(s) test failed
```
Thanks in advance.

**Comment by D3DeFi** [comment]
Same thing happened in #64986 , both of these PRs had previously failed on CI tests resulting in UNSTABLE and were closed and then reopened to rerun CI. Unstable task was [this one](https://github.com/ansible/ansible/blob/98f19c970fe0533de0dc8938eda713c838617945/test/integration/targets/setup_mysql_db/tasks/main.yml#L94) if I am not mistaken (not related to zabbix modules). I don't know how to access older CI builds otherwise I would link failing test directly.

**Comment by mkrizek** [comment]
@Akasurde The PR contains non-module files (tests in this case). Automerge doesn't apply in such case.

**Comment by Akasurde** [comment]
> @Akasurde The PR contains non-module files (tests in this case). Automerge doesn't apply in such case.

Ah! Thanks for information. @D3DeFi @sky-joker ^^

**Comment by D3DeFi** [comment]
oh, ok. Thanks for clarification and merge guys! Will know in the future :)

**Comment by sky-joker** [comment]
Thank you all :)

**Pull request merged**
